components/list/list.js

```diff
@@ -0,0 +1,71 @@
+import './list-item.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+
+class List extends LitElement {
+	static get properties() {
+		return {
+			divider: { type: String }
+		};
+	}
+
+	static get styles() {
+		const layout = css`
+			:host {
+				display: block;
+			}
+			ul {
+				list-style-type: none;
+				padding: 0;
+			}
+			::slotted(*) {
+				display: none;
+			}
+			::slotted(d2l-list-item) {
+				--d2l-list-item-divider-bottom: block;
+				--d2l-list-item-divider-top: block;
+				box-sizing: content-box;
+				display: list-item;
+				margin: 1px 0;
+			}
+		`;
+
+		const specialDividers = css`
+			:host([divider="none"]) ::slotted(d2l-list-item) {
+				--d2l-list-item-divider-bottom: none;
+				--d2l-list-item-divider-top: none;
+			}
+			:host([divider="bottom"]) ::slotted(d2l-list-item) {
+				--d2l-list-item-divider-bottom: block;
+				--d2l-list-item-divider-top: none;
+			}
+			:host([divider="top"]) ::slotted(d2l-list-item) {
+				--d2l-list-item-divider-bottom: none;
+				--d2l-list-item-divider-top: block;
+			}
+			:host([divider="middle"]) ::slotted(d2l-list-item:first-of-type),
+			:host([divider="middle"]) ::slotted(d2l-list-item:last-of-type) {
+				--d2l-list-item-divider-bottom: none;
+				--d2l-list-item-divider-top: none;
+			}
+			:host([divider-extend]) ::slotted(d2l-list-item) {
+				--d2l-list-item-content-padding: 0 18px;
+			}
+			:host(.d2l-list-divider-hover) ::slotted(d2l-list-item:hover) {
```

Comment:
This will probably get replaced with how ever I handle hovering in the next PR.

Mainly, wanted to make sure what I was doing is going to work for selecting or links.